pkg/endpoint/bpf.go

```diff
@@ -783,6 +771,7 @@ func (e *Endpoint) runPreCompilationSteps(regenContext *regenerationContext) (he
 			return false, err
 		}
 
+		e.OnDNSPolicyUpdateLocked(e.owner.GetDNSRules(e.ID))
```

Comment:
The endpoint lock is already held here, but the callee will now also try to grab the lock.

Reply:
Woops, I misread this as `OnDNSPolicyUpdate()`. This is correct.

Reply:
Synced offline. False alarm, but did lead to my latest push where I forgot to add OnDNSPolicyUpdateLocked before the other call to writeHeaderfile.